src/twisted/internet/defer.py

```diff
@@ -583,8 +707,13 @@ def addTimeout(
         self,
         timeout: float,
         clock: IReactorTime,
-        onTimeoutCancel: Optional[Callable[[object, float], object]] = None,
-    ) -> "Deferred[_DeferredResultT]":
+        onTimeoutCancel: Optional[
+            Callable[
+                [Union[_DeferredResultT, Failure], float],
+                Union[_NextDeferredResultT, Failure],
```

Comment:
Maybe this is related to the `addBoth` that I commented about on line 662?

Reply:
Yeeaahh. With that fixed, I get `src/twisted/internet/defer.py:809:55: error: Argument 1 to "addBoth" of "Deferred" has incompatible type "Callable[[_T], _T]"; expected "Callable[[object], Failure]"` and I think I am missing one of the other overloads from `addCallback`

Reply:
OK the type inferencer just kind of gives up here unless you give it this unnecessary help that breaks it in other circumstances. I don't think this is the sort of thing that application code is likely to be doing, so I'll just add the annotation here and move on.  Definitely a mypy bug but any attempt to minify it makes it go away, so we will have to figure out exactly what it is later.